src/test/java/springboot/webapp/usersmanager/services_tests/UserServiceTests.java

```diff
@@ -0,0 +1,144 @@
+
+    User user2 = new User(2, "Sergey", "Danilov", "danilov@gmail.com", Role.ROLE_USER);
+
+    User user3 = new User(3, "Ivan", "Petrov", "petrov@gmail.com", Role.ROLE_USER);
+
+    @Test
+    public void getAll_usersList_ok() throws Exception {
+        List<User> users = new ArrayList<>(Arrays.asList(user1, user2, user3));
+
+        when(userService.getAll()).thenReturn(users);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/users")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(3)));
+    }
+
+    @Test
+    public void getById_existedUser_ok() throws Exception {
+        when(userService.get(user1.getId())).thenReturn(Optional.of(user1));
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/users/1")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", notNullValue()))
+                .andExpect(jsonPath("$.name", is("Sergey")))
+                .andExpect(jsonPath("$.surname", is("Gonchar")));
+    }
+
+    @Test
+    public void getById_nonExistentUser_notFound() throws Exception {
+        when(userService.get(user1.getId())).thenReturn(Optional.of(user1));
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/users/2")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void put_newUser_ok() throws Exception {
+
+        when(userService.put(getUser())).thenReturn(true);
+
+        MockHttpServletRequestBuilder mockHttpServletRequestBuilder = MockMvcRequestBuilders.put("/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(this.mapper.writeValueAsString(getUser()));
+
+        mockMvc.perform(mockHttpServletRequestBuilder)
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", notNullValue()))
+                .andExpect(jsonPath("$", is(true)));
+    }
+
+    @Test
+    public void put_userWithExistedEmail_conflict() throws Exception {
+        when(userService.put(any())).thenThrow(IllegalArgumentException.class);
+
+        String content = "{\"id\" : \"4\",\"name\" : \"Vladislav\",\"surname\" : \"Petrov\",\"role\" : \"ROLE_USER\"}";
+
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.put("/users")
+                .accept(MediaType.APPLICATION_JSON)
+                .content(content)
+                .contentType(MediaType.APPLICATION_JSON);
+        MvcResult result = mockMvc.perform(requestBuilder).andReturn();
+        MockHttpServletResponse response = result.getResponse();
+        assertEquals(HttpStatus.CONFLICT.value(), response.getStatus());
```

Comment:
https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/